src/app/services/counter-analytics.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { CountEventRepositoryService } from '../models/count-event-repository.service';
+import { Counter } from '../models/counter';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CounterAnalyticsService {
+
+  constructor(
+    private countEventRepository: CountEventRepositoryService,
+  ) {}
+
+  getDayOfWeekHistogramData(counter: Counter, eventType: 'positive'|'negative'|'combined'): Array<number> {
```

Comment:
document here that the returned array assumes that first day of the week = sunday

Reply:
also: Instead I would remove the `eventType` input parameter. But just always return an object with:

```javascript
{
  Monday: {
    positive: 1,
    negative: 12,
  },
  Tuesday: {
    positive: 2,
    negative: 3,
  }
  ...
}
```

maybe we could even make this return object a type `CountGroupedByDayOfTheWeek` or so?
This way we avoid this day of the week, `Sunday == array[0]` semantic which I really don't like

Reply:
Good idea. But the keys shall be treated as enumerators not as the text to display in the UI right? Otherwise we will have to deal with i18n in the service.

Reply:
yes. No internationalization in the services. There i'd stick to english only. The transformation should happen at the very last layer in the component before visualization.

Reply:
Implemented returning a better specified type.